Sources/Rownd/Models/AppleSignUpCoordinator.swift

```diff
@@ -0,0 +1,88 @@
+        //Create a request
+        let request         = appleIDProvider.createRequest()
+        //Define the scope of the request
+        request.requestedScopes = [.fullName, .email]
+        //Make the request
+        let authorizationController = ASAuthorizationController(authorizationRequests: [request])
+        
+        //Assingnig the delegates
+        authorizationController.presentationContextProvider = self
+        authorizationController.delegate = self
+        authorizationController.performRequests()
+    }
+    
+    func presentationAnchor(for controller: ASAuthorizationController) -> ASPresentationAnchor {
+        let vc = UIApplication.shared.windows.last?.rootViewController
+        return (vc?.view.window!)!
+    }
+    
+    //If authorization is successfull then this method will get triggered
+    func authorizationController(controller: ASAuthorizationController, didCompleteWithAuthorization authorization: ASAuthorization) {
+        switch authorization.credential {
+        case let appleIDCredential as ASAuthorizationAppleIDCredential:
+            
+            // Create an account in your system.
+            //let userIdentifier = appleIDCredential.user
+            let fullName = appleIDCredential.fullName
+            let email = appleIDCredential.email
+            let identityToken = appleIDCredential.identityToken
+            
+            if let identityToken = identityToken,
+               let urlContent = NSString(data: identityToken, encoding: String.Encoding.ascii.rawValue) {
+                let id_token = urlContent as String
+                Auth.refreshToken(refreshToken: nil, id_token: id_token) { authState in
+                    store.dispatch(SetAuthState(payload: AuthState(accessToken: authState?.accessToken, refreshToken: authState?.refreshToken)))
+                    var userData = store.state.user.data
+                    if let email = email {
+                        userData["email"] = AnyCodable.init(email)
+                    }
+                    if let givenName = fullName?.givenName {
+                        userData["first_name"] = AnyCodable.init(givenName)
+                    }
+                    if let familyName = fullName?.familyName {
+                        userData["last_name"] = AnyCodable.init(familyName)
+                    }
+                    store.dispatch(UserData.save(userData))
+                }
+            } else {
+                logger.trace("apple sign credential alternative")
+            }
+            
+        default:
+            logger.trace("apple sign credential break")
+            break
+        }
+    }
+    
+    //If authorization faced any issue then this method will get triggered
+    func authorizationController(controller: ASAuthorizationController, didCompleteWithError error: Error) {
+        
+        //If there is any error will get it here
+        logger.trace("apple sign credential error")
```

Comment:
we can provide a bit better log message here to help with debugging. something like `logger.error("An error occurred while signing in with Apple. Error: \String(describing: error))")`

Reply:
Logger has been updated